File: app/internal/publisher/template/node/widgets/settings_widget.js
class SettingsWidget {
  constructor(parent, engine) {
    this.parent = parent;
    this.engine = engine;
  }

  async refresh() {
    const status = this.engine.getStatus() || await this.engine.initialize();
    const config = this.engine.config;
    console.log('=== License Information ===');
    console.log(`Status: ${status ? status.status.toUpperCase() : 'UNKNOWN'}`);
    console.log(`Product: ${config.product?.name || '--'}`);
    console.log(`Expiry: ${status?.expires_at || 'N/A'}`);
    console.log(`Plan: ${status?.plan || 'N/A'}`);
    console.log(`Hardware ID: ${status?.hardware_id || '--'}`);
    console.log(`Runtime: Node.js`);
    console.log(`SDK Version: ${config.product?.version || '--'}`);
    console.log('');
    console.log('Actions: (1) Activate  (2) Renew  (3) Replace Device  (4) Open Welcome');
  }

  async openActivation() {
    const { ActivationDialog } = require('../activation');
    const client = this.engine._client;
    const cache = this.engine._cache;
    await new ActivationDialog(client, this.engine.config.product?.name || '', cache).show();
    await this.refresh();
  }

  async openRenewal() {
    const { RenewalDialog } = require('../renewal');
    const key = this.engine.getLicenseKey();
    if (key) {
      await new RenewalDialog(this.engine, key).show();
      await this.refresh();
    }
  }

  async openReplace() {
    const { DeviceReplaceDialog } = require('../device_replace');
    const key = this.engine.getLicenseKey();
    if (key) {
      await new DeviceReplaceDialog(this.engine, key).show();
      await this.refresh();
    }
  }

  async openWelcome() {
    const { WelcomeDialog } = require('../welcome');
    const client = this.engine._client;
    await new WelcomeDialog(client, this.engine.config.product?.name || '').show();
    await this.refresh();
  }
}

module.exports = { SettingsWidget };

File: app/internal/publisher/template/rust/src/widgets/settings_widget.rs
use crate::license_engine::LicenseEngine;
use crate::{ActivationDialog, DeviceReplaceDialog, RenewalDialog, WelcomeDialog};

pub struct SettingsWidget {
    engine: *mut LicenseEngine,
}

impl SettingsWidget {
    pub fn new(engine: &mut LicenseEngine) -> Self {
        Self {
            engine: engine as *mut LicenseEngine,
        }
    }

    pub fn render_info(&mut self) -> String {
        self.refresh();
        let engine = unsafe { &mut *self.engine };
        let mut output = String::new();
        if let Some(ref s) = engine.get_status_cloned() {
            output.push_str(&format!("Status: {}\n", s.status));
            if let Some(ref hw) = s.hardware_id {
                output.push_str(&format!("Hardware ID: {}\n", hw));
            }
            if let Some(ref exp) = s.expires_at {
                output.push_str(&format!("Expiry: {}\n", exp));
            }
            if let Some(ref plan) = s.plan {
                output.push_str(&format!("Plan: {}\n", plan));
            }
        }
        if let Some(ref lk) = engine.license_key {
            output.push_str(&format!("License Key: {}\n", lk));
        }
        output
    }

    pub fn refresh(&mut self) {
        let engine = unsafe { &mut *self.engine };
        engine.initialize();
    }

    pub fn open_activation(&mut self) {
        let engine = unsafe { &mut *self.engine };
        let _ = ActivationDialog::show(engine);
        self.refresh();
    }

    pub fn open_renewal(&mut self) {
        let engine = unsafe { &mut *self.engine };
        if let Some(ref key) = engine.license_key.clone() {
            let _ = RenewalDialog::show(engine, key);
            self.refresh();
        }
    }

    pub fn open_replace(&mut self) {
        let engine = unsafe { &mut *self.engine };
        if let Some(ref key) = engine.license_key.clone() {
            let _ = DeviceReplaceDialog::show(engine, key);
            self.refresh();
        }
    }

    pub fn open_welcome(&mut self) {
        let engine = unsafe { &mut *self.engine };
        let _ = WelcomeDialog::show(engine);
        self.refresh();
    }
}
